### ISSUE TITLE
Needs porting to Python GI API for Gtk

### ISSUE THREAD

**Issue by quozl** [sugarlabs/jigsaw-puzzle-branch#1]
Activity is based on Gtk2 and the older Abiword API.

**Comment by quozl** [comment]
Abiword dependency removed by #2.

**Comment by quozl** [comment]
@zeecoder606 has been working on this ... without letting us know here.  :grin:

https://github.com/zeecoder606/jigsaw-puzzle-branch

**Comment by quozl** [comment]
Saw this on IRC #sugar, but Siddharth did not respond later, and don't know which GitHub user:

```
<siddharthks> Hi, Siddharth Sharma here... I have completely
set up the development environment and most of the activities
on the store are working prefectly
(except for the jigsaw puzzle one). ...
```

Possibly that is this issue.  Activity logs would confirm.

**Comment by quozl** [comment]
@yashagrawal3, you may be interested in a downstream impact of this issue, see [Debian bug #790158](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=790158) which also affects Ubuntu.  Thanks to Jeremy Bicha for the ping.

**Comment by yell0wfl4sh** [comment]
@quozl , Going to add this to activity in GSoC proposal for porting. If urgent, please tell.

**Comment by quozl** [comment]
@yashagrawal3, no, not urgent, if you are presenting any sort of metadata in your proposal, like "reason this activity was selected", it could be "redistributed by Debian and Ubuntu".  As discussed in meeting, there are other reasons, like "redistributed by Fedora and included in SoaS".  Consider also "has strong pedagogical basis" as advised by @walterbender.  Otherwise, as you've found, there's an almost endless supply of Gtk+ 2 activities yet to port.

**Comment by quozl** [comment]
https://github.com/sugarlabs/jigsaw-puzzle-branch/pull/5 help wanted.
